### PR TITLE
Key Vault test proxy migration

### PR DESCRIPTION
**Pull request by vcolin7**

### CODE REVIEW

**Comment by billwert** [comment]
can you rebase this so you're not pulling in the change that's already in main?

File: sdk/keyvault/azure-security-keyvault-administration/pom.xml [hunk 1]
```diff
@@ -89,7 +89,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core-test</artifactId>
-      <version>1.15.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
+      <version>1.16.0-beta.1</version> <!-- {x-version-update;com.azure:azure-core-test;current} -->
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-administration/pom.xml:92]: Need the latest released core-test dep.

File: sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/KeyVaultSecretsUserAgentPropertiesTest.java [hunk 2]
```diff
@@ -4,9 +4,10 @@
 package com.azure.security.keyvault.secrets;
 
 import com.azure.core.util.CoreUtils;
-import java.util.Map;
 import org.junit.jupiter.api.Test;
 
+import java.util.Map;
```
Comment (billwert) [inline sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/KeyVaultSecretsUserAgentPropertiesTest.java:9]: unused?

File: sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java [hunk 3]
```diff
@@ -63,6 +74,19 @@ protected List<HttpPipelinePolicy> getPolicies() {
...
+                List<TestProxySanitizer> customSanitizers = new ArrayList<>();
+                customSanitizers.add(new TestProxySanitizer("token", null, "REDACTED", TestProxySanitizerType.BODY_KEY));
+                interceptorManager.addSanitizers(customSanitizers);
+            }
+        } else {
+            credential = new MockTokenCredential();
+
+            List<TestProxyRequestMatcher> customMatchers = new ArrayList<>();
+            customMatchers.add(new BodilessMatcher());
+            customMatchers.add(new CustomMatcher().setExcludedHeaders(Collections.singletonList("Authorization")));
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java:88]: "Authorization" was this still needed to be excluded after MockTokenCredential?

File: sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java [hunk 4]
```diff
@@ -81,14 +87,15 @@ protected String getTestName() {
     }
 
     void beforeTestSetup() {
+        KeyVaultCredentialPolicy.clearCache();
```
Comment (samvaity) [inline sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java:90]: We don't need this for the admin client Tests?

**Review by billwert — APPROVED** [review]
lgtm!

**Review by samvaity — APPROVED** [review]
other changes, lgtm!

File: sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java (hunk 4 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java:90]: Yes, we do. Let me make sure it's there.

File: sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java (hunk 3 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java:88]: I think I added this before moving onto `MockTokenCredential`. I can give a shot to removing it.

File: sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/KeyVaultSecretsUserAgentPropertiesTest.java (hunk 2 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/KeyVaultSecretsUserAgentPropertiesTest.java:9]: No, we definitely use it.

File: sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java (hunk 4 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateClientTestBase.java:90]: We set it on `KeyVaultAdministrationClientTestBase`.

File: sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java (hunk 3 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultAdministrationClientTestBase.java]: Looks like we needed it :v

**Pull request merged**